src/Http/Exception/DataValidationException.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * This source file is proprietary and part of Rebilly.
+ *
+ * (c) Rebilly SRL
+ *     Rebilly Ltd.
+ *     Rebilly Inc.
+ *
+ * @see https://www.rebilly.com
+ */
+
+namespace Rebilly\Http\Exception;
+
+use Exception;
+
+/**
+ * Class DataValidationException.
+ */
+final class DataValidationException extends UnprocessableEntityException
+{
+    private $validationErrors = [];
+
+    public function __construct(array $content = [], $message = '', $code = 0, Exception $previous = null)
+    {
+        if (isset($content['invalidFields']) && is_array($content['invalidFields'])) {
+            foreach ($content['invalidFields'] as $field => &$errors) {
+                if (is_array($errors)) {
+                    foreach ($errors as &$error) {
+                        $error = $error['message'] ?? '';
```

Comment:
Wonder why do you stringify the error array?

Reply:
So, because forexample, we can use         
```
} catch (DataValidationException $e) {
     $this->addErrors($e->getErrors());
```
but yii has : 
```
	public function addErrors($errors)
	{
		foreach($errors as $attribute=>$error)
		{
			if(is_array($error))
			{
				foreach($error as $e)
					$this->addError($attribute, $e);
			}
			else
				$this->addError($attribute, $error);
		}
	}
```
two nested array only, and here will be an error, as we have in some our code.

Reply:
There is wrong assumption that this SDK is written for Yii.

Reply:
Yes, of course not for Yii , but there are different usages. 
Someone expects them as: 
```
Array
(
    [couponIds.0] => Array
        (
            [0] => Array
                (
                    [message] => Coupon Code must exist
                )
        )
)
```
but another like : 
```
Array
(
    [couponIds.0] => Array
        (
            [0] => Coupon Code must exist
        )
)
```
(with out extra nested array for `message` key ) (how we use it too in case of yii)

what do you suggest?

Reply:
ok, maybe let it be as it is? and this is the responsibility of the who will use it and how.  that is their concern.

Reply:
ok, maybe you are right, extra transformations removed